zetaclient/chains/solana/signer/outbound_tracker_reporter.go

```diff
@@ -0,0 +1,90 @@
+package signer
+
+import (
+	"context"
+	"time"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+)
+
+const (
+	// SolanaTransactionTimeout is the timeout for waiting for an outbound to be confirmed
+	// Transaction referencing a blockhash older than 150 blocks will expire and be rejected by Solana.
+	SolanaTransactionTimeout = 2 * time.Minute
+)
+
+// reportToOutboundTracker launch a go routine with timeout to check for tx confirmation;
+// it reports tx to outbound tracker only if it's confirmed by the Solana network.
+func (signer *Signer) reportToOutboundTracker(
+	ctx context.Context,
+	zetacoreClient interfaces.ZetacoreClient,
+	chainID int64,
+	nonce uint64,
+	txSig solana.Signature,
+	logger zerolog.Logger,
+) {
+	// set being reported flag to avoid duplicate reporting
+	alreadySet := signer.Signer.SetBeingReportedFlag(txSig.String())
+	if alreadySet {
+		logger.Info().
+			Msgf("reportToOutboundTracker: outbound %s for chain %d nonce %d is being reported", txSig, chainID, nonce)
+		return
+	}
+
+	// launch a goroutine to monitor tx confirmation status
+	go func() {
+		defer func() {
+			signer.Signer.ClearBeingReportedFlag(txSig.String())
+		}()
+
+		start := time.Now()
```

Comment:
Can we use `retry` package here?

Reply:
Not sure if it supports a time range based retry. But I'll check it out.